src/app/services/api/color-scheme.service.ts

```diff
@@ -0,0 +1,64 @@
+import {Injectable, Renderer2, RendererFactory2} from '@angular/core';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class ColorSchemeService {
+
+    private renderer: Renderer2;
+    private colorScheme: string;
+    // Define prefix for clearer and more readable class names in scss files
+    private colorSchemePrefix = 'color-scheme-';
+
+    constructor(rendererFactory: RendererFactory2) {
+        // Create new renderer from renderFactory, to make it possible to use renderer2 in a service
+        this.renderer = rendererFactory.createRenderer(null, null);
+    }
+
+    _detectPrefersColorScheme() {
```

Comment:
i like the leading _. could mark these private as well

Reply:
I'd rather we not have leading underscores. It's a remnant from javascript days to indicate that method is private. In Typescript you can just use `private` keyword for this.

It's not even recommended in javascript: https://github.com/airbnb/javascript#naming--leading-underscore